front/components/workspace/ProviderManagementModal.tsx

```diff
@@ -65,19 +67,31 @@ export function ProviderManagementModal({
     },
     [setProviderStates]
   );
+
+  const handleSave = useCallback(() => {
+    const activeProviders = MODEL_PROVIDER_IDS.filter(
```

Comment:
Why use a useCallback?

Reply:
True, function is quite straightforward and probably wouldn't significantly benefit from memoization. Will ditch it